nltk/translate/chrf_score.py

```diff
@@ -110,28 +132,59 @@ def corpus_chrf(list_of_references, hypotheses, min_len=1, max_len=6, beta=3.0):
     :type max_len: int
     :param beta: the parameter to assign more importance to recall over precision
     :type beta: float
+    :param ignore_whitespace: ignore whitespace characters in scoring
+    :type ignore_whitespace: bool
     :return: the sentence level CHRF score.
     :rtype: float
     """
 
-    assert len(list_of_references) == len(hypotheses), "The number of hypotheses and their references should be the same"
+    assert len(references) == len(hypotheses), (
+        "The number of hypotheses and their references should be the same")
+
+    # Keep f-scores for each n-gram order separate
+    n_gram_fscores = {k: [] for k in range(min_len, max_len + 1)}
```

Comment:
It's not necessary to pre-populate, hashmaps with empty array as values in Python. If the aim is for optimization, then an fixed size array should be used as the values not the default Python list that is dynamic.

Reply:
I'm fairly sure I would get a `KeyError` if I try to append to a list corresponding to a nonexistent key. One option would be to use `collections.defaultdict` but this seems a bit like overkill here.

Reply:
Not an overkill. `defaultdict` seems like the perfect usage here =)

Reply:
Ok, done.